pkg/controller/user-cluster-controller-manager/resources/reconciler.go

```diff
@@ -687,9 +687,11 @@ func (r *reconciler) reconcileValidatingWebhookConfigurations(ctx context.Contex
 		creators = append(creators, csimigration.ValidatingwebhookConfigurationReconciler(data.caCert.Cert, metav1.NamespaceSystem, resources.VsphereCSIMigrationWebhookConfigurationWebhookName))
 	}
 
-	if r.cloudProvider == kubermaticv1.VSphereCloudProvider || r.cloudProvider == kubermaticv1.NutanixCloudProvider || r.cloudProvider == kubermaticv1.OpenstackCloudProvider ||
-		r.cloudProvider == kubermaticv1.DigitaloceanCloudProvider {
-		creators = append(creators, csisnapshotter.ValidatingSnapshotWebhookConfigurationReconciler(data.caCert.Cert, metav1.NamespaceSystem, resources.CSISnapshotValidationWebhookConfigurationName))
+	if !data.cluster.Spec.DisableCSIDriver {
+		if r.cloudProvider == kubermaticv1.VSphereCloudProvider || r.cloudProvider == kubermaticv1.NutanixCloudProvider || r.cloudProvider == kubermaticv1.OpenstackCloudProvider ||
+			r.cloudProvider == kubermaticv1.DigitaloceanCloudProvider {
+			creators = append(creators, csisnapshotter.ValidatingSnapshotWebhookConfigurationReconciler(data.caCert.Cert, metav1.NamespaceSystem, resources.CSISnapshotValidationWebhookConfigurationName))
+		}
```

Comment:
Only this one change is not sufficient:
- Ensure that no other resource for CSI drivers is created when CSI driver is disabled
- Just his change wouldn't work, if i disable CSI driver after cluster creation. So you have to make sure that these CSI resources that are managed via code are removed when CSI driver is disabled for a cluster -- for example https://github.com/kubermatic/kubermatic/blob/main/pkg/controller/user-cluster-controller-manager/resources/reconciler.go#L1339-L1347